include/zephyr/drivers/video-controls.h

```diff
@@ -31,42 +38,91 @@ extern "C" {
+#define VIDEO_CTRL_CLASS_CODEC			0x00990900 /**< Stateful codec controls */
+#define VIDEO_CTRL_CLASS_CAMERA			0x009a0900 /**< Camera class controls */
+#define VIDEO_CTRL_CLASS_FM_TX			0x009b0900 /**< FM Modulator controls */
+#define VIDEO_CTRL_CLASS_FLASH			0x009c0900 /**< Camera flash controls */
+#define VIDEO_CTRL_CLASS_JPEG			0x009d0900 /**< JPEG-compression controls */
+#define VIDEO_CTRL_CLASS_IMAGE_SOURCE		0x009e0900 /**< Image source controls */
+#define VIDEO_CTRL_CLASS_IMAGE_PROC		0x009f0900 /**< Image processing controls */
+#define VIDEO_CTRL_CLASS_DV			0x00a00900 /**< Digital Video controls */
+#define VIDEO_CTRL_CLASS_FM_RX			0x00a10900 /**< FM Receiver controls */
+#define VIDEO_CTRL_CLASS_RF_TUNER		0x00a20900 /**< RF tuner controls */
+#define VIDEO_CTRL_CLASS_DETECT			0x00a30900 /**< Detection controls */
+#define VIDEO_CTRL_CLASS_CODEC_STATELESS	0x00a40900 /**< Stateless codecs controls */
+#define VIDEO_CTRL_CLASS_COLORIMETRY		0x00a50900 /**< Colorimetry controls */
+#define VIDEO_CTRL_CLASS_VENDOR			0x00ff0900 /**< Vendor-specific class controls */
 /**
  * @}
  */
 
 /**
- * @name Generic class control IDs
+ * @name Base class control IDs
  * @{
  */
-/** Mirror the picture horizontally */
-#define VIDEO_CID_HFLIP			(VIDEO_CTRL_CLASS_GENERIC + 0)
-/** Mirror the picture vertically */
-#define VIDEO_CID_VFLIP			(VIDEO_CTRL_CLASS_GENERIC + 1)
+#define VIDEO_CID_BRIGHTNESS			(VIDEO_CTRL_CLASS_BASE + 0)
+#define VIDEO_CID_CONTRAST			(VIDEO_CTRL_CLASS_BASE + 1)
+#define VIDEO_CID_SATURATION			(VIDEO_CTRL_CLASS_BASE + 2)
+#define VIDEO_CID_EXPOSURE			(VIDEO_CTRL_CLASS_BASE + 17)
+#define VIDEO_CID_GAIN				(VIDEO_CTRL_CLASS_BASE + 19)
+#define VIDEO_CID_HFLIP				(VIDEO_CTRL_CLASS_BASE + 20)
+#define VIDEO_CID_VFLIP				(VIDEO_CTRL_CLASS_BASE + 21)
+#define VIDEO_CID_WHITE_BALANCE_TEMPERATURE	(VIDEO_CTRL_CLASS_BASE + 26)
 /**
  * @}
  */
 
 /**
- * @name Camera class control IDs
+ * @name Camera class controls IDs
  * @{
  */
-#define VIDEO_CID_CAMERA_EXPOSURE	(VIDEO_CTRL_CLASS_CAMERA + 0)
-#define VIDEO_CID_CAMERA_GAIN		(VIDEO_CTRL_CLASS_CAMERA + 1)
-#define VIDEO_CID_CAMERA_ZOOM		(VIDEO_CTRL_CLASS_CAMERA + 2)
-#define VIDEO_CID_CAMERA_BRIGHTNESS	(VIDEO_CTRL_CLASS_CAMERA + 3)
-#define VIDEO_CID_CAMERA_SATURATION	(VIDEO_CTRL_CLASS_CAMERA + 4)
-#define VIDEO_CID_CAMERA_WHITE_BAL	(VIDEO_CTRL_CLASS_CAMERA + 5)
-#define VIDEO_CID_CAMERA_CONTRAST	(VIDEO_CTRL_CLASS_CAMERA + 6)
-#define VIDEO_CID_CAMERA_COLORBAR	(VIDEO_CTRL_CLASS_CAMERA + 7)
-#define VIDEO_CID_CAMERA_QUALITY	(VIDEO_CTRL_CLASS_CAMERA + 8)
+#define VIDEO_CID_ZOOM_ABSOLUTE			(VIDEO_CTRL_CLASS_CAMERA + 13)
+/**
+ * @}
+ */
+
+/**
+ * @name JPEG class control IDs
+ * @{
+ */
+#define VIDEO_CID_JPEG_COMPRESSION_QUALITY	(VIDEO_CTRL_CLASS_JPEG + 3)
+/**
+ * @}
+ */
+
+/**
+ * @name Image Processing class control IDs
+ * @{
+ */
+#define VIDEO_CID_TEST_PATTERN			(VIDEO_CTRL_CLASS_IMAGE_PROC + 3)
+/**
+ * @}
+ */
+
+/**
+ * @name Compatibility control IDs
```

Comment:
I think this should be removed. Why do we need to keep compatibility while all can be changed at once ? especially the changes are not much as currently only the ov2640 has used controls IDs in upstream. On-going PRs can rebase on this if it is merged first

Reply:
I was thinking about 3rd-party projects. I do not have any myself though.
It is also a trivial fix on 3rd-party projects to migrate to the new IDs!